Library/Homebrew/cmd/options.rb

```diff
@@ -32,13 +35,40 @@ def options
       puts_options Formula.to_a.sort
     elsif args.installed?
       puts_options Formula.installed.sort
+    elsif !args.command.nil?
+      path = Commands.path(args.command)
+      odie "Unknown command: #{args.command}" unless path
+      cmd_options = if cmd_parser = CLI::Parser.from_cmd_path(path)
+        cmd_parser.processed_options.map { |x| [x.second.presence || x.first, x[3]] }
```

Comment:
Might be worth using variables here for first, second, third, the `map do` block form so it's obvious what's being used and why.